util/tune/current_settings.rb

```diff
@@ -0,0 +1,366 @@
+    output = `#{command}`
+
+    if output
+      # See the docs link in the description above
+      # The default used in PE is the number of CPUs - 1, expressed as $::processorcount - 1.
+      # One instance is the minimum value and four instances is the maximum value.
+      num_cores = output.to_i
+      value = [[(num_cores - 1), MIN_DEFAULT_JRUBIES].max, MAX_DEFAULT_JRUBIES].min.to_s
+    else
+      value = NA
+    end
+
+  end
+
+  value
+end
+
+# Returns the java args for the specified file
+#
+# @author Bill Claytor
+#
+# @param [String] file The file to search for java args
+#
+# @return [string] The java args
+#
+# @example
+#   args = get_java_args(file)
+#
+# TODO: grep for java args
+def get_java_args(file)
+  path = if File.exist? "/etc/debian_version"
+           "/etc/defaults/#{file}"
+         else
+           "/etc/sysconfig/#{file}"
+         end
+
+  value = if File.exist? path
+            File.open(path).grep(/Xmx/)[0].split('"')[1]
+          else
+            NA
+          end
+
+  value
+end
+
+# Returns the value for the following setting:
+# "puppet_enterprise::master::puppetserver::reserved_code_cache"
+#
+# Uses the 'ReservedCodeCacheSize' parameter in the 'pe-puppetserver' file
+#
+# @author Bill Claytor
+#
+# @return [string] The reserved_code_cache value
+#
+# @example
+#   value = puppetserver_reserved_code_cache
+#
+#
+def puppetserver_reserved_code_cache
+  get_java_args("pe-puppetserver").match(/XX:ReservedCodeCacheSize=\K[^\s]+/) || NA
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::console::java_args'
+#
+# Uses the 'pe-console-services' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = console_java_args
+#
+def console_java_args
+  get_java_args("pe-console-services")
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::master::java_args'
+#
+# Uses the 'pe-puppetserver' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = master_java_args
+#
+def master_java_args
+  get_java_args("pe-puppetserver")
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::orchestrator::java_args'
+#
+# Uses the 'pe-orchestration-services' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = zzz
+#
+def orchestrator_java_args
+  get_java_args("pe-orchestration-services")
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::puppetdb::java_args'
+#
+# Uses the 'pe-puppetdb' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = puppetdb_java_args
+#
+def puppetdb_java_args
+  get_java_args("pe-puppetdb")
+end
+
+# Returns the value for the specified parameter in the specified config file
+# Optionally accepts a list of exclusions to avoid collisions (i.e. 'work_mem' matching 'autovacuum_work_mem', etc)
```

Comment:
The use of excluded words was added to satisfy the case of `database_work_mem` which is looking for the parameter 'work_mem'. This also matches 'autovacuum_work_mem' and 'maintenance_work_mem'. Excluding the words you don't want to match for specific cases seemed like the easiest way to handle this rather than creating a more complicated regular expression for all cases.
```
def database_work_mem
  exclusions = /(maintenance|autovacuum)/
  get_postgres_parameter("work_mem", exclusions)
end
```
If methods for other parameters need to be added in the future and have similar collision issues they can just pass their own exclusions rather than modifying the common regular expression used by all cases.